auction-server/src/watcher.rs

```diff
@@ -0,0 +1,79 @@
+use {
+    crate::{
+        auction::ChainStore,
+        server::{
+            EXIT_CHECK_INTERVAL,
+            SHOULD_EXIT,
+        },
+        state::{
+            Store,
+            SvmChainUpdate,
+        },
+    },
+    anyhow::{
+        anyhow,
+        Result,
+    },
+    futures::StreamExt,
+    solana_sdk::{
+        commitment_config::CommitmentConfig,
+        hash::Hash,
+    },
+    solana_transaction_status::TransactionDetails,
+    std::{
+        str::FromStr,
+        sync::{
+            atomic::Ordering,
+            Arc,
+        },
+    },
+};
+
+pub async fn run_svm_watcher_loop(store: Arc<Store>, chain_id: String) -> Result<()> {
+    let chain_store = store
+        .chains_svm
+        .get(&chain_id)
+        .ok_or(anyhow!("Chain not found: {}", chain_id))?;
+
+    let mut exit_check_interval = tokio::time::interval(EXIT_CHECK_INTERVAL);
+
+    let ws_client = chain_store.get_ws_client().await?;
+    let (mut stream, _) = ws_client
+        .block_subscribe(
+            solana_client::rpc_config::RpcBlockSubscribeFilter::All,
+            Some(solana_client::rpc_config::RpcBlockSubscribeConfig {
+                encoding:                          None,
+                transaction_details:               Some(TransactionDetails::None), // we don't need transaction data
+                show_rewards:                      Some(false),
+                max_supported_transaction_version: None,
+                commitment:                        Some(CommitmentConfig::finalized()),
```

Comment:
Lets use `confirmed` while this is not yet fixed in quick node. Or If we have another endpoint which is gonna work with finalized config.

Reply:
done